### PR TITLE
move db query off main thread, fix group self-add

### LINKED ISSUE #4168: possible to add dupe contact to group

### PR DESCRIPTION
**Pull request by mcginty**
Fixed adding yourself to a group, as well as us doing DB queries on the main thread to check for push eligibility.

Also fixed a regression where updating a group said "New group" in title.

fixes #4168

### CODE REVIEW

**Comment by moxie0** [comment]
in 3.4.0

File: res/values/strings.xml [hunk 1]
```diff
@@ -233,8 +233,9 @@
     <string name="GroupCreateActivity_menu_create_title">Create group</string>
     <string name="GroupCreateActivity_creating_group">Creating %1$s&#8230;</string>
     <string name="GroupCreateActivity_updating_group">Updating %1$s...</string>
-    <string name="GroupCreateActivity_cannot_add_non_push_to_existing_group">Cannot add non-Signal contacts to an existing Signal group</string>
+    <string name="GroupCreateActivity_cannot_add_non_push_to_existing_group">Couldn\'t add %1$s because they\'re not a Signal user.</string>
```
Comment (McLoo) [inline res/values/strings.xml]: As this is shown per recipient now, shouldn't
they're
be something singular?

Comment (mcginty) [inline res/values/strings.xml]: "they" can be used as a gender-neutral singular

Comment (turtlekiosk) [inline res/values/strings.xml]: 👍🏽

**Pull request closed without merging**